service/controller/cluster.go

```diff
@@ -64,6 +66,8 @@ func NewCluster(config ClusterConfig) (*Cluster, error) {
 		c := certs.Config{
 			K8sClient: config.K8sClient,
 			Logger:    config.Logger,
+
+			WatchTimeout: 5 * time.Second,
```

Comment:
We go with a shorter timeout for the certs watcher. Works really nice and the reconciliation loops are more short lived.